### PR TITLE
[core] Bump antlr4 from 4.7 to 4.7.2

### PR DESCRIPTION
**Pull request by alecharp**
## Describe the PR

A security fix is included in this version. See https://github.com/antlr/antlr4/issues/2069.

## Ready?

- [x] Added unit tests for fixed bug/feature
- [ ] Passing all unit tests
- [ ] Complete build `./mvnw clean verify` passes (checked automatically by travis)
- [x] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pom.xml [hunk 1]
```diff
@@ -97,7 +97,7 @@
         <pmd.plugin.version>3.13.0</pmd.plugin.version>
         <ant.version>1.10.8</ant.version>
         <javadoc.plugin.version>3.2.0</javadoc.plugin.version>
-        <antlr.version>4.7</antlr.version>
+        <antlr.version>4.7.1</antlr.version>
```
Comment (adangel) [inline pom.xml:100]:
```suggestion
        <antlr.version>4.7.2</antlr.version>
```

The latest 4.7.x version is 4.7.2 -> https://search.maven.org/artifact/org.antlr/antlr4-runtime

**Review by adangel — COMMENTED** [review]
Thanks for the PR!
Did you encounter a specific issue you want to solve? Because just updating this dependency is probably not enough in order to run PMD with a security manager. We have a couple of `System.getenv` and `System.getProperty` calls as well. There might be more method calls, especially for reading files, using reflection, ...

**Comment by alecharp** [comment]
@adangel `pmd-core` is used in a Jenkins plugin and this update was required when the plugin was analyzed by a security tool.

**Comment by adangel** [comment]
@alecharp Thanks for info, I'll merge it.

**Pull request merged**